packages/material-ui/src/styles/experimentalStyled.js

```diff
@@ -2,6 +2,18 @@ import styled from '@material-ui/styled-engine';
 import { propsToClassKey } from '@material-ui/styles';
 import defaultTheme from './defaultTheme';
 
+function isEmpty(obj) {
+  let result = true;
+
+  Object.keys(obj).forEach((key) => {
+    if (obj.hasOwnProperty(key)) {
```

Comment:
question: For what environment is this required? By spec [`Object.keys()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys) returns all own enumerable properties so this condition should always be true. If there's some browser/polyfill that doesn't implement it then we should document it.

Reply:
You are correct I should just check the length of `Object.keys()`, will refactor it